docs/multicluster/user-guide.md

```diff
@@ -630,6 +633,98 @@ Normally, `podCIDRs` should be the value of `kube-controller-manager`'s
 will not be enabled. If you use `kubectl edit` to edit the ConfigMap, then you
 need to restart the `antrea-mc-controller` Pod to load the latest configuration.
 
+## Multi-cluster Stretched NetworkPolicy
+
+Network Policy can be enforeced on cross-cluster traffic in a ClusterSet in the following ways:
+
+### Egress Stretched NetworkPolicy
+Restricting Pod egress traffic to backends of a Multi-cluster Service (which can be on the
+same cluster of the Pod or on a different cluster) is supported by Antrea-native policy's
+[toServices](../antrea-network-policy.md#toservices-egress-rules) feature. To define such
+policy, simply put the imported Service name and namespace in the `toServices` field of an
+Antrea-native policy:
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: acnp-drop-tenant-to-secured-mc-service
+spec:
+  priority: 1
+  tier: securityops
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          role: tenant
+  egress:
+    - action: Drop
+      toServices:
+        - name: antrea-mc-secured-service  # an imported Multi-cluster Service
+          namespace: svcNamespace
+```
+Note that the scope of policy's appliedTo field will still be restricted to the cluster
+where the policy is created in. To enforce such policy for all `role=tenant` Pods in the
+entire ClusterSet, use the [ClusterNetworkPolicy Replication](#multi-cluster-clusternetworkpolicy-replication)
+feature mentioned in the previous section, and set the `clusterNetworkPolicy` field of 
+the ResourceExport to the `acnp-drop-tenant-to-secured-mc-service` spec above. Such
+replication should only be performed by ClusterSet admins, who have clearance of creating
+ClusterNetworkPolicies in all clusters of a ClusterSet.
+
+### Ingress Stretched NetworkPolicy
+Alternatively, Antrea-native policies now support selecting ingress peers at the ClusterSet
+scope (since v1.10.0). Policy rules can be created to enforce security postures on ingress
+traffic from all member clusters in a ClusterSet:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: drop-tenant-access-to-admin-namespace
+spec:
+  appliedTo:
+  - namespaceSelector:
+      matchLabels:
+        role: admin
+  priority: 1
+  tier: securityops
+  ingress:
+  - action: Deny
+    from:
+    # Select all Pods in role=tenant namespaces in the ClusterSet
+    - scope: clusterSet
```

Comment:
Question - do you think applying "scope" to the whole policy makes the implementation and also consumption simpler (even that means ClusterSet policy does not support egress rules)?

Reply:
Reminder

Reply:
IMO keeping the `scope` field in the NetworkPolicy peer makes the semantics of this toggle much clearer. Scope=ClusterSet for the entire policy is not accurate at all, because the policy only applies to local cluster workloads, in addition to the egress not supported side you mentioned.

Reply:
While I can agree with your point on "semantic", I feel in reality probably users will not mix both regular rules and stretched rules in one policy; or say separating the rules to two different policies will not impact user experience, and may be easier to manage/understand in some cases.

Again, could you answer my question about implementation complexity - if we have a separate policy type for stretched policies, will the implementation be simpler? If yes, I would consider that approach. I also know other security policy implementation goes that way.

Reply:
We have definitely entertained with the alternative of having a new CRD, see this slide https://docs.google.com/presentation/d/1LzikjVY8M_dfwtnvJxyravb7MxlntQQozfsvfC-LUb4/edit#slide=id.g13af9598368_0_161 (which we presented on the community meetings). 

I think the challenge here is not related to implementation of the feature itself, but how the other parts would come together. If we have a new CRD, we need to introduce Tier and Priority to that CRD as well, otherwise the relative precedence between that new policy type and ACNP/ANP become ambiguous. So now, 1. users will have to check yet another resource to understand all security rules applied to the cluster 2. they won't be able to use the `antctl` cmd and other tools that we provide (out of the box) to visualize relative priorities between multi-cluster and single cluster rules 3. additional logic needs to be added for supporting Tier RBAC for the new CRD for commercial users. For these reasons, we decided to go with the 'add new field to ACNP/ANP' approach by then.

That being said, as we mentioned in the slides, there's obviously upsides in having a separate CRD. If we keep this feature in ACNP/ANP though, I would not recommend moving `scope` to the policy level, as the meaning of the policy will be too confusing. Right now the downside of the current approach is if user want to select multiple ClusterSet-level peers (I agree mixing cluster and ClusterSet level peers does not sound like a plausible usecase), they would have to put `scope: clusterSet` toggle on for each peer.

/cc @GraysonWu @tnqn @antoninbas @salv-orlando

Reply:
Sorry, I do not mean to have a new CRD, but still mean putting the "scope" field at the policy level (not per rule level).

Reply:
My bad, I misunderstood. Then no, there will be no implementation implications. It will purely be API design choice.